ethcore/src/miner/miner.rs

```diff
@@ -1306,31 +1306,39 @@ impl miner::MinerService for Miner {
 
 	// Note used for external submission (PoW) and internally by sealing engines.
 	fn submit_seal(&self, block_hash: H256, seal: Vec<Bytes>) -> Result<SealedBlock, Error> {
-		let result =
-			if let Some(b) = self.sealing.lock().queue.get_used_if(
-				if self.options.enable_resubmission {
-					GetAction::Clone
-				} else {
-					GetAction::Take
-				},
-				|b| &b.header.bare_hash() == &block_hash
-			) {
-				trace!(target: "miner", "Submitted block {}={} with seal {:?}", block_hash, b.header.bare_hash(), seal);
-				b.lock().try_seal(&*self.engine, seal).or_else(|e| {
-					warn!(target: "miner", "Mined solution rejected: {}", e);
-					Err(Error::PowInvalid.into())
-				})
-			} else {
+		let action = if self.options.enable_resubmission {
+			GetAction::Clone
+		} else {
+			GetAction::Take
+		};
+
+		let block = self.sealing.lock().queue
+			.get_used_if(action, |b| &b.header.bare_hash() == &block_hash)
+			.ok_or_else(|| {
 				warn!(target: "miner", "Submitted solution rejected: Block unknown or out of date.");
-				Err(Error::PowHashInvalid.into())
-			};
+				Error::PowHashInvalid
+			})?;
 
-		result.and_then(|sealed| {
-			let n = sealed.header.number();
-			let h = sealed.header.hash();
-			info!(target: "miner", "Submitted block imported OK. #{}: {}", Colour::White.bold().paint(format!("{}", n)), Colour::White.bold().paint(format!("{:x}", h)));
-			Ok(sealed)
-		})
+		trace!(target: "miner", "Submitted block {hash}={bare_hash} with seal {seal:?}",
+			   hash = block_hash,
```

Comment:
mixed tabs and spaces, a proper way to indent multiline function call with tabs is 
```rust
call(
	arg1,
	arg2,
);
```